coalib/bearlib/aspects/root.py

```diff
@@ -15,9 +15,9 @@ class Root(aspectbase, metaclass=aspectclass):
     >>> from coalib.bearlib.aspects import Taste
 
     >>> @Root.subaspect
-    ... class Formatting:
+    ... class Format:
```

Comment:
why are you renaming this?

Reply:
related to https://github.com/coala/coala/pull/5492#issuecomment-396062741

Reply:
I am pretty confident with your summary of the problem there.

What I suspect is that the doctests are not cleaning up after themselves.

Please create a new branch for investigating this.  Revert this rename so the problem re-occurs, and then change the pytest configuration to disable doctests.

If the error goes away when the doctests are disabled, we have 100% confirmed the doctest are polluting the Root aspect.

Then you can create an issue about this.
It is a multi-faceted issue.
One part to investigate is what happens when two subaspects are registered with the same name..?
Why isnt there an error when the second one (i.e. doctests and the real one) is registered?

The real solution should include for the doctest in this module to delete the `Formatting` that they created. (I am not sure how that can be done -- deleting aspects may not have been planned, so you may need to delete objects from a dictionary somewhere.)
Then we know how to un-do damage caused by doctest.

But a workaround is to create a separate PR "related to" the issue , which gives this subaspect a different name.  `Format` isnt a good new name.  Give it a crazy name, so we will always recognise it when it appears in logs unexpectedly. ;-)

Reply:
@jayvdb error goes away when the doctests are disabled in pytest. Now the tests are passing (although ultimately failing due to incomplete coverage).

I have created the issue here https://github.com/coala/coala/issues/5518.

About the multi-aspect part, I think you need to see https://github.com/coala/coala/blob/936a5d82a7ce6df5fac30ef39f08f2d2e53e3205/coalib/bearlib/aspects/__init__.py#L99. We already have `MultipleAspectFoundError` but it's worth noting that it's not raised for the two `Formatting` aspects (one of doctests and the actual one). This needs to be investigated.

I like the idea of the workaround, issue lies here https://github.com/coala/coala/blob/master/coalib/bearlib/aspects/root.py#L18